src/KernelAbstractions.jl

```diff
@@ -62,10 +67,63 @@ abstract type Event end
 import Base.wait
 
 # TODO
-function async_copy! end
 # function register end
 # function unregister end
 
+# Functions for async_copy
+function pin!(a)
+  ad = Mem.register(Mem.Host, pointer(a), sizeof(a))
+  finalizer(_ -> Mem.unregister(ad), a)
+end
+
+function recordevent(stream)
+  event = CuEvent(CUDAdrv.EVENT_DISABLE_TIMING)
+  CUDAdrv.record(event, stream)
+  return CudaEvent(event)
+end
+
+function async_copy!(A,B)
+    unsafe_copyto!(pointer(A), pointer(B), length(A))
+end
+
+if has_cuda_gpu()
+    function async_copy!(A::CuArray, B::CuArray; stream=CuDefaultStream(),
```

Comment:
The stream chosen here should come from the `stream pool` in `backends/cuda.jl`.

also probably means we want `async_copy!(::CUDA, )`